File: code-clone/src/app/shared/models/file-inputs/Snippet.ts
import {CodeReference} from "./CodeReference";
import {CodeInput} from "./CodeInput";
import {InputType} from "./CodeReference";

export class Snippet implements CodeReference, CodeInput{
  contents: string;
  type: InputType;
  fileName: string;

  constructor(fileName?: string, fileContents?: string) {
    this.type = InputType.SNIPPET;
    this.fileName=fileName;
    this.contents=fileContents;
  }




}

File: code-clone-detector/src/main/java/com/clonedetector/codeclonedetector/model/Snippet.java
package com.clonedetector.codeclonedetector.model;

public class Snippet {
    public String type;
    public String fileName;
    public String contents;

    public String toString() {
        return "Snippet{" +
                "type=" + type + '\n' +
                ", fileName=" + fileName + '\n' +
                ", contents=" + contents + '\n' +
                '}';
    }
}
